Add bookings

DanielDaze opened the pull request:
Добавлены логика бронирования и комментарии

DanielDaze committed:
- add schema.sql
- add jpa notifications for models

DanielDaze committed:
- add UserRepository.java

DanielDaze committed:
- ItemRepository WIP

DanielDaze committed:
- item tests passed

DanielDaze committed:
- BookingRepository WIP

DanielDaze committed:
- BookingRepository WIP

DanielDaze committed:
- BookingRepository WIP

DanielDaze committed:
- add BookingInfo

DanielDaze committed:
- add BookingInfo

DanielDaze committed:
- Booking logic done

DanielDaze committed:
- comments WIP

DanielDaze committed:
- comments WIP

DanielDaze committed:
- comments WIP

DanielDaze committed:
- postman tests clear

DanielDaze committed:
- postman tests clear

DanielDaze committed:
- db fix

DanielDaze committed:
- db fix

johnskyi commented on src/main/java/ru/practicum/shareit/booking/Booking.java:
```diff
@@ -1,20 +1,40 @@
 /**
  * TODO Sprint add-bookings.
  */
 @Data
```
при добавлении этой аннотации необходимо исключить из toString который она содержит все поля со связями, для избежания бросков исключений. Поправь пожалуйста сразу во всех местах

johnskyi commented on src/main/java/ru/practicum/shareit/booking/BookingRepository.java:
```diff
@@ -0,0 +1,39 @@
+import java.util.List;
+import java.util.Optional;
+
+public interface BookingRepository extends JpaRepository<Booking, Long> {
```
@Repository аннотацию не забывай

johnskyi commented on src/main/java/ru/practicum/shareit/booking/BookingServiceImpl.java:
```diff
@@ -0,0 +1,124 @@
+
+@Service
+@RequiredArgsConstructor
+public class BookingServiceImpl implements BookingService {
```
кажется вы уже должны были пройти про транзакции. Если да, то пожалуйста обеспечь транзакционность всех етодов обращения к базе во всех сервисах. Также необходимо логировать все операции с сущностями

DanielDaze committed:
- @Repository annotations added
- @ToString.Exclude annotations added
- @Transactional annotations added
- added logging on return
